common/reviews/api/rush-lib.api.md

```diff
@@ -376,7 +403,6 @@ export class RushConfiguration {
     readonly repositoryUrl: string | undefined;
     readonly rushJsonFile: string;
     readonly rushJsonFolder: string;
-    readonly rushLinkJsonFilename: string;
```

Comment:
This will be a breaking API change. I kinda doubt anyone is actually using this, but it is technically a breaking change.

This might be okay, but we should document this well in the changelog.

Reply:
Why should we remove it if it still works for Yarn?

Reply:
This change removes `rush-link.json` entirely, from Yarn, NPM, and PNPM installs. I don't see a purpose keeping two different systems of local dependency tracking (one being the new API, the other being this generated file) when we only need one. The dependency graph shouldn't depend on whether or not you're using workspaces, and should be fully attainable by looking at the Rush configuration as well as the `package.json` files.